transcom/mymove
# B 19804 int

## [B-19804](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A946250)

## Summary

Downstream to the 19800 zip/city lookup functionality. This adds the capability for Office users when editing the Customer Profile address or adding/editing shipment addresses.

Is there anything you would like reviewers to give additional scrutiny?

I don't like using essentially the same handler 6 times in the ShipmentForm.jsx file. The only difference between them is the field names that are being used to in the setValues method. If anybody knows of a way to use a string of the name of the object or a way to pass in the object being used in LocationSearchBox back to ShipmentForm I would greatly appreciate the assistance.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the application as an Office user
2. Select a move and then edit the shipment
3. Verify you can use the zip/city lookup to add to the address
4. Make sure you can save the new address
5. Add a new shipment and verify you can use the zip/city lookup and add an address (This applies for secondary and tertiary address for pickup and destination)
6. Edit the Customer Profile and use the zip/city lookup to change the Customer address for current and backup and verify the address saves

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
